autoinstall_templates/sample.seed

```diff
@@ -117,6 +117,10 @@ d-i pkgsel/include string ntp ssh wget
 # Note: options passed to the installer will be added automatically.
 d-i debian-installer/add-kernel-opts string $kernel_options_post
 
+# Grub settings
+d-i grub-installer/grub2_instead_of_grub_legacy boolean true
+d-i grub-installer/bootdev string /dev/[sv]da
```

Comment:
okay, I was able to reproduce on Debian PXE (only tested with 8.8.0 with the netboot initrd.tar.gz replacement hack).
But according to https://www.debian.org/releases/stable/example-preseed.txt
It is possible to set:
d-i grub-installer/bootdev string default

Given that this is only needed on debian, I would suggest to use
#if $breed == "debian"
d-i grub-installer/bootdev string default
#end if

It should be possible to specify the arches at some point .
I'm not sure which arches are using grub2 over lilo or extlinux on debian. At least on fedora, extlinux is used by default on armhfp.
#if ( $breed == "debian") and ( "arm" not in $arch )